app/controllers/classified_listings_controller.rb

```diff
@@ -96,6 +92,32 @@ def dashboard
 
   private
 
+  def create_listing(purchaser, cost)
+    # this will 500 for now if they don't belong in the org
+    authorize @classified_listing, :authorized_organization_poster? if @classified_listing.organization_id.present?
+
+    @classified_listing.bumped_at = Time.current
+    @classified_listing.published = true
+
+    if @classified_listing.save
```

Comment:
It seems like bumping a listing (saving) and spending credits (marking as spent) should be inside a transaction. What do you think about it?

Reply:
It's a good call, there shouldn't be a situation where the listing is on the DB but the credits haven't been subtracted. Same with all the other operations that spend credits.
Let me see what's the best way to go about that without revolutionizing the code